proxy.go

```diff
@@ -0,0 +1,100 @@
+// A simple proxy that multiplexes a unix socket connection
+//
+// Copyright 2017 HyperHQ Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+package main
+
+import (
+	"flag"
+	"io"
+	"net"
+	"strings"
+	"sync"
+
+	"github.com/golang/glog"
+	"github.com/hashicorp/yamux"
+)
+
+func serv(servConn io.ReadWriteCloser, proto, addr string) error {
+	session, err := yamux.Client(servConn, nil)
+	if err != nil {
+		glog.Errorf("fail to create yamux client: %s", err)
+		return err
+	}
+
+	// serving connection
+	l, err := net.Listen(proto, addr)
+	if err != nil {
+		glog.Errorf("fail to listen on %s:%s: %s", proto, addr, err)
+		return err
+	}
+	defer l.Close()
+
+	for {
+		conn, err := l.Accept()
+		if err != nil {
+			glog.Errorf("fail to accept new connection: %s", err)
+			return err
+		}
+		stream, err := session.Open()
```

Comment:
Could you please put the whole stream opening, stream copy and sync logic into a dedicated function and call a go routine on it for each accept? I think this would make the code more readable.